File: back-end/src/main/java/com/huyphan/models/enums/OrderDirection.java
package com.huyphan.models.enums;

import com.fasterxml.jackson.annotation.JsonValue;

/** Order direction. */
public enum OrderDirection {
	Asc("asc"), Desc("desc");

	private String value;

	private OrderDirection(String value) {
		this.value = value;
	}

	@JsonValue
	public String getValue() {
		return value;
	}
}

File: front-end/src/app/core/enums/order-direction.ts
/** Order direction. */
export enum OrderDirection {
  Asc = 'asc',
  Desc = 'desc',
}

export namespace OrderDirection {
  const TO_READABLE_MAP: Readonly<Record<OrderDirection, string>> = {
    [OrderDirection.Asc]: 'Ascending',
    [OrderDirection.Desc]: 'Descending',
  };

  /**
   * Maps to enum value into a human-readable representation of it.
   * @param direction Enum value to map into readable one.
   */
  export function toReadable(direction: OrderDirection): string {
    return TO_READABLE_MAP[direction];
  }
}
